fossunited/fossunited/doctype/grants_funding_directory/grants_funding_directory.py

```
@@ -0,0 +1,259 @@
+        frappe.log_error(f"Validation error: {e}")
+        return {"error": True, "message": f"Validation error: {str(e)}"}
+
+
+@frappe.whitelist()
+def refresh_funding_data(docname: str) -> dict:
+    """Refresh funding data by re-fetching and validating JSON."""
+    try:
+        doc = frappe.get_doc("Grants Funding Directory", docname)
+        doc._fetch_and_validate_json()
+        doc.save()
+        return {"success": True, "message": "Funding data refreshed successfully"}
```

Ruff (0.14.5)

Consider moving this statement to an `else` block

(TRY300)